crates/settlement-clients/ethereum/src/lib.rs

```diff
@@ -0,0 +1,86 @@
+pub mod clients;
+pub mod config;
+pub mod conversion;
+pub mod types;
+
+use alloy::{
+    network::EthereumWallet,
+    primitives::{Address, U256},
+    providers::ProviderBuilder,
+    signers::local::PrivateKeySigner,
+};
+use async_trait::async_trait;
+use color_eyre::Result;
+use config::EthereumSettlementConfig;
+use conversion::{slice_slice_u8_to_vec_u256, slice_u8_to_u256};
+use mockall::{automock, predicate::*};
+use settlement_client_interface::{SettlementClient, SettlementVerificationStatus};
+use std::sync::Arc;
+use utils::env_utils::get_env_var_or_panic;
+
+use crate::clients::interfaces::validity_interface::StarknetValidityContractTrait;
+use crate::clients::StarknetValidityContractClient;
+
+pub const ENV_PRIVATE_KEY: &str = "ETHEREUM_PRIVATE_KEY";
+
+#[allow(dead_code)]
+pub struct EthereumSettlementClient {
+    core_contract_client: StarknetValidityContractClient,
+}
+
+#[automock]
+#[async_trait]
+impl SettlementClient for EthereumSettlementClient {
+    /// Should register the proof on the base layer and return an external id
+    /// which can be used to track the status.
+    #[allow(unused)]
+    async fn register_proof(&self, proof: Vec<u8>) -> Result<String> {
+        Ok("external_id".to_string())
+    }
+
+    /// Should be used to update state on core contract when DA is done in calldata
+    #[allow(unused)]
+    async fn update_state_calldata(
+        &self,
+        program_output: Vec<Vec<u8>>,
+        onchain_data_hash: Vec<u8>,
+        onchain_data_size: usize,
+    ) -> Result<String> {
+        let program_output: Vec<U256> = slice_slice_u8_to_vec_u256(&program_output);
+        let onchain_data_hash: U256 = slice_u8_to_u256(&onchain_data_hash);
+        self.core_contract_client.update_state(program_output, onchain_data_hash, U256::from(0)).await?;
+        Ok("TODO".to_string())
+    }
+
+    /// Should be used to update state on core contract when DA is in blobs/alt DA
+    #[allow(unused)]
+    async fn update_state_blobs(&self, program_output: Vec<Vec<u8>>, kzg_proof: Vec<u8>) -> Result<String> {
+        let program_output: Vec<U256> = slice_slice_u8_to_vec_u256(&program_output);
+        self.core_contract_client.update_state_kzg(program_output, kzg_proof).await?;
+        Ok("TODO".to_string())
+    }
+
+    /// Should verify the inclusion of the state diff in the DA layer and return the status
+    #[allow(unused)]
+    async fn verify_inclusion(&self, external_id: &str) -> Result<SettlementVerificationStatus> {
+        // TODO: not implemented yet
```

Comment:
Added

Reply:
I'd say this would not be the responsibility of this job and probably the worker, like 1. check verify inclusion 2. if pending, wait a bit 3. recheck verify inclusion?

Reply:
Added!

Reply:
I am not sure what was added actually. We still don't wait for 2/3rd confirmation right?

Reply:
I added a `wait_for_tx_finality` function that's waiting for a tx to be confirmed. 

> for 2/3rd confirmation 
What do you mean by this? That would be `CONFIRMED ON L2` if the settlement layer is starknet? What about eth?

Reply:
So an ETH block is only final when 2/3rd of the stake votes on it which takes around 12mins. Initially, I was thinking we should wait for this confirmation. However, looking at the stats - https://etherscan.io/uncles, it looks like we haven't had any uncle blocks since merge. So in the worst case, if we do have reorgs (uncle blocks), our txs revert (and we get the fees back) and our future code might break. However, the chances of this seem to be really low that we can assume we will handle this manually if and when this happens